BaseLib/FileTools.cpp

```diff
@@ -79,12 +79,12 @@ std::size_t findLastDot(std::string const& path)
 std::string dropFileExtension(std::string const& filename)
 {
 	// Look for dots in filename.
-	const std::size_t p = findLastDot(filename);
+	const std::string::size_type p = findLastDot(filename);
```

Comment:
I don't like auto all that much because it obfuscates the real type (especially when the real type is not straightforward as in this case).